oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/endpoint/NimbusJwtClientAuthenticationParametersConverter.java

```diff
@@ -210,4 +231,59 @@ private JWK getJwk() {
 
 	}
 
+	/**
+	 * A context that holds client authentication-specific state and is used by
+	 * {@link NimbusJwtClientAuthenticationParametersConverter} when attempting to build
+	 * the JSON Web Token (JWS) client assertion.
+	 *
+	 * @param <T> the type of {@link AbstractOAuth2AuthorizationGrantRequest}
+	 * @since 5.7
+	 */
+	public static final class JwtClientAuthenticationContext<T extends AbstractOAuth2AuthorizationGrantRequest> {
```

Comment:
I think you can remove `extends AbstractOAuth2AuthorizationGrantRequest` since the bound is defined in outer class.